### PR TITLE
AWS StepFunctions services are generated using ACK Crossplane pipeline

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

`Activity` and `StateMachine` resources added by using ACK Crossplane generation pipeline.

Fixes https://github.com/crossplane/provider-aws/issues/450

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Manually with YAMLs in `examples` folder.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/sfn/v1alpha1/zz_enums.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	StateMachineStatus_SDK_ACTIVE   StateMachineStatus_SDK = "ACTIVE"
+	StateMachineStatus_SDK_DELETING StateMachineStatus_SDK = "DELETING"
+)
+
+type StateMachineType string
+
+const (
+	StateMachineType_STANDARD StateMachineType = "STANDARD"
+	StateMachineType_EXPRESS  StateMachineType = "EXPRESS"
+)
```
Comment (krishchow) [inline apis/sfn/v1alpha1/zz_enums.go:112]: Where are these enums being used/referenced?

File: apis/sfn/v1alpha1/zz_state_machine.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+// StateMachineParameters defines the desired state of StateMachine
+type StateMachineParameters struct {
+	// Region is which region the StateMachine will be created.
+	// +kubebuilder:validation:Required
+	Region string `json:"region"`
+	// The Amazon States Language definition of the state machine. See Amazon States
+	// Language (https://docs.aws.amazon.com/step-functions/latest/dg/concepts-amazon-states-language.html).
+	// +kubebuilder:validation:Required
+	Definition *string `json:"definition"`
```
Comment (krishchow) [inline apis/sfn/v1alpha1/zz_state_machine.go:35]: Should this be a *string or a string?

File: apis/sfn/v1alpha1/zz_state_machine.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	//    * wildcard characters ? *
+	//
+	//    * special characters " # % \ ^ | ~ ` $ & , ; : /
+	//
+	//    * control characters (U+0000-001F, U+007F-009F)
+	//
+	// To enable logging with CloudWatch Logs, the name should only contain 0-9,
+	// A-Z, a-z, - and _.
+	// +kubebuilder:validation:Required
+	Name *string `json:"name"`
```
Comment (krishchow) [inline apis/sfn/v1alpha1/zz_state_machine.go:59]: Should this be a *string or a string?

File: apis/sfn/v1alpha1/zz_types.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type ActivityListItem struct {
+	ActivityARN *string `json:"activityARN,omitempty"`
+
+	CreationDate *metav1.Time `json:"creationDate,omitempty"`
+
+	Name *string `json:"name,omitempty"`
+}
```
Comment (krishchow) [inline apis/sfn/v1alpha1/zz_types.go:31]: None of the structs in this file have comments for documentation, do they need documentation (i.e., will any of these be exposed to users via CRDs)?

**Review by krishchow — CHANGES_REQUESTED** [review]
Just a few comments/notes

File: apis/sfn/v1alpha1/custom_types.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+
+// CustomActivityParameters includes custom additional fields for ActivityParameters.
+type CustomActivityParameters struct{}
+
+// CustomStateMachineParameters includes custom additional fields for StateMachineParameters.
+type CustomStateMachineParameters struct {
+	// RoleARN is the ARN for the IAMRole.
+	// +immutable
+	RoleARN *string `json:"roleArn,omitempty"`
```
[the remark this answers is not in the document]
Comment (muvaf) [reply apis/sfn/v1alpha1/custom_types.go:28]: From scheme perspective, it's optional; you can create the CR without it. But it has to exist for AWS calls. So, I added a sentence explaining that.

File: apis/sfn/v1alpha1/zz_enums.go (hunk 1 above)
Comment (muvaf) [reply apis/sfn/v1alpha1/zz_enums.go:112]: Some of them are used and some of them aren't. Technically we can skip generating those but they are available in the pipeline and I though they could be useful for people who import these as packages to know what values are available programmatically.

File: apis/sfn/v1alpha1/zz_state_machine.go (hunk 2 above)
Comment (muvaf) [reply apis/sfn/v1alpha1/zz_state_machine.go:35]: Yeah, it should be a `string` but ACK copies the types from AWS Go SDK directly so we get everything as pointers.

File: apis/sfn/v1alpha1/zz_types.go (hunk 4 above)
Comment (muvaf) [reply apis/sfn/v1alpha1/zz_types.go:31]: Comments are available in ACK only for certain fields and I couldn't find a good way to inject them properly. For now, I opted to get back to ACK to fix them later on.

File: apis/sfn/v1alpha1/custom_types.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (krishchow) [reply apis/sfn/v1alpha1/custom_types.go]: Got it, okay, thank you :)

File: apis/sfn/v1alpha1/zz_enums.go (hunk 1 above)
Comment (krishchow) [reply apis/sfn/v1alpha1/zz_enums.go:112]: Makes sense

File: apis/sfn/v1alpha1/zz_types.go (hunk 4 above)
Comment (krishchow) [reply apis/sfn/v1alpha1/zz_types.go:31]: Ah okay, I see, thanks for clarifying

**Review by krishchow — APPROVED** [review]
LGTM!

**Pull request merged**